docs/sources/ocimetrics.md

```diff
@@ -0,0 +1,73 @@
+# Oracle Cloud Infrastructure Metrics (OCIMetrics) source for knative eventing
+
+This event source collects metrics data from the [Oracle Cloud](https://cloud.oracle.com).
+
+## Prerequisites
+
+An Oracle Cloud account with permissions to inspect and read metrics for the
+Oracle Cloud Infrastructure (OCI) compartment is required.  In addition, the
+account must have the account API secret added.
+
+For additional information on how to create an API key and associate it with
+your Oracle Cloud user, go to: https://docs.cloud.oracle.com/en-us/iaas/Content/API/Concepts/apisigningkey.htm#Required_Keys_and_OCID
+
+### Adding the Oracle Cloud Secret
+
+Three pieces of information are required for the Oracle Cloud:
+1. API Private Key used for signing the request
+1. API Private Key passphrase to decrypt the key
+1. API Key's fingerprint to identify which key to use on the Oracle Cloud end
+
+From the Navigation Menu, click `Secrets`, then click the `ADD SECRET` button on
+the upper right corner. A list of secrets will be presented on the right side.
+Click the `Oracle` menu item.  Add the private key ensuring the `-----BEGIN RSA PRIVATE KEY-----`
+and `-----END RSA PRIVATE KEY-----` are included.  Finally, click the `SAVE` button.
+
+![Adding an Oracle Cloud API Secret](../images/ocimetrics-source/oracle-add-secret.png)
+
+The secret can now be used for both the OCIMetrics Source and Oracle Targets.
+
+## Deploying an instance of the Source
+
+### Creating the Source
+Open the Bridge creation screen and add a source of type `OCIMetrics`.
+
+![Adding an OCI Metric](../images/ocimetrics-source/create-bridge-1.png)
+
+In the source creation form, provide a unique name and broker. These are used by
+Triggermesh to uniquely identify the source and where to send the events to.
+
+For the Oracle Cloud specific information, provide the following information:
+* Oracle tenancy using the Oracle Cloud ID (OCID)
+* Oracle username as an OCID
+* Oracle Cloud region where the metrics should be pulled from
+
+For the metrics specific information:
+* Metrics namespace such as `oci_computeagent` or `oci_vcn`
+* The metrics query based on [MQL](https://docs.cloud.oracle.com/en-us/iaas/Content/Monitoring/Reference/mql.htm)
+
+For details on how to write a query, consult the
+[Oracle Cloud Monitoring Overview](https://docs.cloud.oracle.com/en-us/iaas/Content/Monitoring/Concepts/monitoringoverview.htm)
+
+![Adding OCI Metric Details](../images/ocimetrics-source/create-bridge-2.png)
+
+After clicking the `SAVE` button, you will be taken back to the Bridge editor.
+Continue to add the targets, and then submit the bridge.
+
+![Bridge overview](../images/ocimetrics-source/create-bridge-3.png)
+
+## Event types
+
+The CloudEvent type is:
+
+    com.oracle.cloud.monitoring
+
+The CloudEvent source of the form:
+
+    ocimetrics/<namespace>/<source-name>
```

Comment:
Didn't we add "sources.triggermesh.io" to the source type? I can't remember whether it happened in the PR, but I remember mentioning it.

Reply:
No as per https://github.com/triggermesh/event-sources/blob/fa8926c0ddafa3d338eb448c390f8211f3261872/pkg/apis/sources/v1alpha1/ocimetricssource_lifecycle.go#L66. Thought the triggermesh specific event types would be applicable towards targets instead of sources.

Reply:
To be fair we don't have a pattern for source types. We use URLs where applicable, ARN or equivalent for cloud resources, a custom string with name/namespace otherwise.

It's out of scope here of course, but for my own curiosity, would OCI provide a unique project ID or something Oracle-ish we could use as the source attribute instead of leaking k8s details?

Reply:
That it doesn't.  As to the CloudEvent type, the payload does match what Oracle publishes and is emitted verbatim so it makes sense there. As to the CloudEvent source, given that the OCIMetrics source is a user defined query to begin with, we would need something that can be tied to a user defined instance of the source.

Now if we want the CloudEvent source to resemble something out of the knative source community such as:
`/apis/v1/namespaces/<namespace>/ocimetrics/<source-name>` then we should create an issue against the source to fix this.

Reply:
No complain at all about the event type, it's fully qualified and describes what information it contains 👌 

Regarding the source: true, it's a query, I forgot about that. I guess the source is fine as it is then.